lib/Sema/CSBindings.cpp

```diff
@@ -1102,8 +1102,6 @@ bool TypeVariableBinding::attempt(ConstraintSystem &cs) const {
       } else if (srcLocator->getAnchor() &&
                  isExpr<ObjectLiteralExpr>(srcLocator->getAnchor())) {
         fix = SpecifyObjectLiteralTypeImport::create(cs, dstLocator);
-      } else if (srcLocator->isKeyPathRoot()) {
-        fix = SpecifyKeyPathRootType::create(cs, dstLocator);
```

Comment:
This was the right location for this fix.

Reply:
The problem is that this fix is only going to be recorded if the solver attempts to bind the key path root type variable. That doesn't happen when a type variable is a non-representative, in this case, it only attempts binding the generic argument type variable. So the options I thought was let this here, and record a fix on `simplifyKeyPathConstraint` only for non-representatives or record everything there ...

Reply:
I've adjusted to maintain this :)